synapse/links/ssl.py

```diff
@@ -14,10 +14,23 @@
 class Socket(s_socket.Socket):
 
     def send(self, byts):
+        '''
+        Send bytes on the socket.
+
+        Args:
+            byts (bytes): The bytes to send
+
+        Returns:
+            int: The sent byte count (or None) on fini()
+        '''
         try:
-            return s_socket.Socket.send(self, byts)
+            return self.sock.send(byts)
```

Comment:
The problem here was that s_socket.Socket's send was catching the ssl.SSLWantRead/WriteError and returning None, so the ssl socket would just return None instead of 0.

Reply:
in addition to returning None, it was closing the SslSocket prematurely